api/scpca_portal/test/utils/test_input_bucket_s3_key_info.py

```diff
@@ -0,0 +1,89 @@
+from pathlib import Path
+
+from django.test import TestCase
+
+from scpca_portal.enums import FileFormats, Modalities
+from scpca_portal.utils import InputBucketS3KeyInfo
+
+
+class InputBucketS3KeyInfoTest(TestCase):
+    def test_id_parsing(self):
+        project_file = Path("SCPCP999999/file.txt")
+        project_file_s3_key_info = InputBucketS3KeyInfo(project_file)
+        self.assertEqual(project_file_s3_key_info.project_id, "SCPCP999999")
+        self.assertIsNone(project_file_s3_key_info.sample_id)
+        self.assertIsNone(project_file_s3_key_info.library_id)
+        self.assertTrue(project_file_s3_key_info.is_project_file)
+
+        library_file = Path("SCPCP999999/SCPCS999999/SCPCL999999_file.txt")
+        library_file_s3_key_info = InputBucketS3KeyInfo(library_file)
+        self.assertEqual(library_file_s3_key_info.project_id, "SCPCP999999")
+        self.assertEqual(library_file_s3_key_info.sample_id, "SCPCS999999")
+        self.assertEqual(library_file_s3_key_info.library_id, "SCPCL999999")
+        self.assertFalse(library_file_s3_key_info.is_project_file)
+
+    def test_is_merged(self):
+        merged_file = Path("SCPCP000001/merged/SCPCP999999_merged.txt")
+        merged_file_s3_key_info = InputBucketS3KeyInfo(merged_file)
+        self.assertTrue(merged_file_s3_key_info.is_merged)
+
+        non_merged_file = Path("SCPCP999999/SCPCS999999/SCPCL999999_file.txt")
+        non_merged_file_s3_key_info = InputBucketS3KeyInfo(non_merged_file)
+        self.assertFalse(non_merged_file_s3_key_info.is_merged)
+
+    def test_modalities(self):
+        single_cell_file = Path("SCPCP999999/SCPCS999999/SCPCL999999_file.txt")
+        spatial_file = Path("SCPCP999999/SCPCS999999/SCPCL999999_spatial/SCPCL999999_file.txt")
+        cite_seq_file = Path("SCPCP999999/SCPCS999999/SCPCL999999_file_adt.h5ad")
+        bulk_file = Path("SCPCP999999/bulk/SCPCP999999_bulk.txt")
+
+        single_cell_file_s3_key_info = InputBucketS3KeyInfo(single_cell_file)
+        self.assertIn(Modalities.SINGLE_CELL, single_cell_file_s3_key_info.modalities)
+        self.assertNotIn(Modalities.SPATIAL, single_cell_file_s3_key_info.modalities)
+        self.assertNotIn(Modalities.CITE_SEQ, single_cell_file_s3_key_info.modalities)
+        self.assertNotIn(Modalities.BULK_RNA_SEQ, single_cell_file_s3_key_info.modalities)
```

Comment:
```suggestion
        self.assertIn(Modalities.SINGLE_CELL, single_cell_file_s3_key_info.modalities)
        self.assertEqual(1, len(single_cell_file_s3_key_info.modalities))
```

I think this is a bit more concise and can be applied to the others.